src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,43 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public LocalDate getLocalDate(String date) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        return LocalDate.parse(date, formatter);
+    }
+
+    public boolean isValidDate(LocalDate startDate, LocalDate endDate, String currentDate) {
+        LocalDate localDate = new SalaryInfo().getLocalDate(currentDate);
+        return (localDate.isAfter(startDate.minusDays(1))
+            && localDate.isBefore(endDate.plusDays(1)));
+    }
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        SalaryInfo salaryInfo = new SalaryInfo();
+        LocalDate startLocalDate = salaryInfo.getLocalDate(dateFrom);
+        LocalDate endLocalDate = salaryInfo.getLocalDate(dateTo);
+        int[] totalSalaries = new int[names.length];
+
+        for (String datum : data) {
+            String[] splittedData = datum.split(" ");
+            for (int i = 0; i < names.length; i++) {
+                if (names[i].equals(splittedData[1]) && salaryInfo.isValidDate(startLocalDate,
+                        endLocalDate, splittedData[0])) {
+                    totalSalaries[i] +=
+                        Integer.parseInt(splittedData[2]) * Integer.parseInt(splittedData[3]);
```

Comment:
It's hard to understand what is stored under arrayName[someIndex] without knowing the input format, it's always better to extract such expressions in separate variables with proper names to give some context on what is actually going on